daemon/logger/loggerutils/logfile.go

```diff
@@ -168,6 +254,19 @@ func (w *LogFile) ReadLogs(config logger.ReadConfig, watcher *logger.LogWatcher)
 	defer func() {
 		for _, f := range files {
 			f.Close()
+			fileName := f.Name()
+			if strings.HasSuffix(fileName, tmpLogfileSuffix) {
+				w.filesRefCounter.Mu.Lock()
```

Comment:
Can we do all the locking, counter changes in the refcounter object?

Reply:
Now Refcounter only count the tmp log files, so I think It's better to keep a single responsibility. :-)

Reply:
Right but we can do something like `counter.Reference(fileName)` and `counter.Dereference(filename)` rather than dealing with the internals of it here.

Reply:
Hum, I seemly hadn't got what you meant clearly :P, will do. Thanks! :-)

Reply:
@cpuguy83 Done, Please take another look, tks.